### ISSUE TITLE
Can't set float on mobile toolbar button (in my app)

### ISSUE THREAD

**Issue by aerwin** [maqetta/maqetta#3512]
Using Chrome on Mac going to Maqetta.org...

Using the HTML file below:
1. Drag toolbar button onto the heading on the main view (view with id `mainView`)
2. Use props palette to try and change `float` attribute to "right" and get error dialog that I'm trying to edit read-only theme

```
<!DOCTYPE html>
<html>
<head>
<meta charset="utf-8"/>
<title>Untitled</title>
<script type="text/javascript" src="lib/dojo/dojo/dojo.js" data-dojo-config="'parseOnLoad':true,'async':true,'packages':[{'name':'gridx','location':'../gridx'},{'name':'clipart','location':'../../clipart'},{'name':'maqettaSamples','location':'../../../samples'},{'name':'maqetta','location':'../../maqetta'},{'name':'shapes','location':'../../shapes'},{'name':'zazl','location':'../../zazl'},{'name':'widgets','location':'../../custom'}]"></script>
<script type="text/javascript">
require([
  "dijit/dijit",
  "dojo/parser",
  "maqetta/space",
  "maqetta/AppStates",
  "dojox/mobile",
  "dojox/mobile/deviceTheme",
  "dojox/mobile/compat",
  "dojo/hash",
  "dojox/mobile/SpinWheel",
  "dojox/mobile/SpinWheelDatePicker",
  "dojox/mobile/TextArea",
  "dojox/io/xhrScriptPlugin",
  "dojo/data/ItemFileReadStore",
  "dojox/mobile/EdgeToEdgeDataList"
]);
</script>
<style>@import "themes/claro/claro.css";@import "app.css";
</style>
<script type="text/javascript" src="app.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=no"></meta>
</head>
<body class="claro" data-maq-device="iphone" data-maq-flow-layout="true" data-maq-ws="collapse" id="myapp" data-maq-appstates="{}">
 <div data-dojo-type="dojox.mobile.View" id="mainView" style="height: 100%;" selected="selected">
  <h1 label="Weight" data-dojo-type="dojox.mobile.Heading">
 </h1>
  <span data-dojo-type="dojo.data.ItemFileReadStore" id="ItemFileReadStore_2" jsId="ItemFileReadStore_2" data="{&quot;items&quot;:[{&quot;label&quot;:&quot;150&quot;,&quot;moveTo&quot;:&quot;detailsView&quot;},{&quot;label&quot;:&quot;152&quot;,&quot;moveTo&quot;:&quot;detailsView&quot;}]}"></span>
  <ul data-dojo-type="dojox.mobile.EdgeToEdgeDataList" store="ItemFileReadStore_2" query="{&quot;label&quot;:&quot;*&quot;}"></ul>
</div>
 <div data-dojo-type="dojox.mobile.View" id="detailsView" style="height: 100%;">
<h1 label="Details" data-dojo-type="dojox.mobile.Heading" moveTo="mainView" back="Home"></h1>
<div data-dojo-type="dojox.mobile.RoundRect">
   <div data-dojo-type="dojox.mobile.SpinWheel" style="padding: 0px; width: 278px; height: 200px; padding-left: 5px;">
     <div data-dojo-type="dojox.mobile.SpinWheelSlot" labels="0,1,2,3,4,5,6,7,8,9" value="2" style="width: 30%;"></div>
     <div data-dojo-type="dojox.mobile.SpinWheelSlot" labels="0,1,2,3,4,5,6,7,8,9" value="A" style="width: 30%;"></div>
     <div data-dojo-type="dojox.mobile.SpinWheelSlot" labels="0,1,2,3,4,5,6,7,8,9" value="A" style="width: 30%;"></div>
   </div>
 </div>
  <ul data-dojo-type="dojox.mobile.RoundRectList">
    <li data-dojo-type="dojox.mobile.ListItem" moveTo="detailsView_Date">
      Date</li>
    <li data-dojo-type="dojox.mobile.ListItem">
      Notes</li>
  </ul>
</div>
 <div data-dojo-type="dojox.mobile.View" id="detailsView_Notes" style="height: 100%;">
   <h1 label="Notes" data-dojo-type="dojox.mobile.Heading" back="Details" moveTo="detailsView"></h1>
   <div data-dojo-type="dojox.mobile.RoundRect" shadow="false">
     <textarea data-dojo-type="dojox.mobile.TextArea" style="width: 100%; height: 124px;"></textarea>
   </div>
 </div>
 <div data-dojo-type="dojox.mobile.View" id="detailsView_Date" style="height: 100%;">
  <h1 label="Date" data-dojo-type="dojox.mobile.Heading" back="Details" moveTo="detailsView"></h1>
  <div data-dojo-type="dojox.mobile.RoundRect">
   <div data-dojo-type="dojox.mobile.SpinWheelDatePicker"></div>
 </div>
</div>
</body>
</html>
```

What's really strange is that if I start from scratch I can do it successfully:
1. New mobile app
2. Drag view onto page
3. Drag heading onto page
4. Drag mobile toolbar button onto heading
5. In props palette, try to change `float` to right and get error that I'm using a read-only theme

**Comment by JonFerraiolo** [comment]
-> @billreed63 

I'm pretty sure this is a metadata problem, maybe a regression when updating Dojo 1.7 to 1.8 where possibly some of the  CSS class selectors have changed in the Dojo css files. 

This fix might be something as follows. This is current theme metadata for ToolBarButton:

```
            "rootSelectors": [
                ".mblToolBarButton"
            ],
```

It might work with Dojo 1.8 if we change to look like this:

```
            "rootSelectors": [
                ".mblToolBarButton",
                ".mblHeading .mblToolBarButton"
            ],
```

but there might be an additional complexity (and maybe a bug in Dojo) because Dojo changes the DOM structure for the widget if you turn arrows on in widget-specific properties, which causes the CSS selectors for CSS properties to change on us depending on widget-specific properties. @billreed63 and @JonFerraiolo may need to have a conversation about this.

**Comment by JonFerraiolo** [comment]
@aerwin" In the meantime while waiting for this fix, you can go into source view and type `style="float:right"` for the ToolBarButton.

**Comment by billreed63** [comment]
Fixed by commit 7604daf7535da6949c24b93353df6eb6c3c78d8c
